dotCMS/src/main/java/com/dotcms/ai/client/openai/OpenAIClient.java

```diff
@@ -0,0 +1,166 @@
+
+    public static OpenAIClient get() {
+        return INSTANCE.get();
+    }
+
+    private OpenAIClient() {
+        lastRestCall = new ConcurrentHashMap<>();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public AIProvider getProvider() {
+        return AIProvider.OPEN_AI;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public <T extends Serializable> void sendRequest(final AIRequest<T> request, final OutputStream output) {
+        final JSONObjectAIRequest jsonRequest = AIClient.useRequestOrThrow(request);
+        final AppConfig appConfig = jsonRequest.getConfig();
+
+        AppConfig.debugLogger(
+                OpenAIClient.class,
+                () -> String.format(
+                        "Posting to [%s] with method [%s]%s  with app config:%s%s  the payload: %s",
+                        jsonRequest.getUrl(),
+                        jsonRequest.getMethod(),
+                        System.lineSeparator(),
+                        appConfig.toString(),
+                        System.lineSeparator(),
+                        jsonRequest.payloadToString()));
+
+        if (!appConfig.isEnabled()) {
+            AppConfig.debugLogger(OpenAIClient.class, () -> "App dotAI is not enabled and will not send request.");
+            throw new DotAIAppConfigDisabledException("App dotAI config without API urls or API key");
+        }
+
+        final JSONObject payload = jsonRequest.getPayload();
+        final String modelName = Optional
+                .ofNullable(payload.optString(AiKeys.MODEL))
+                .orElseThrow(() -> new DotAIModelNotFoundException("Model is not present in the request"));
+        // TODO: pr-split -> uncomment this line
+        //final Tuple2<AIModel, Model> modelTuple = appConfig.resolveModelOrThrow(modelName, jsonRequest.getType());
+        final Tuple2<AIModel, Model> modelTuple = Tuple.of(null, null);
+        final AIModel aiModel = modelTuple._1;
+
+        if (!modelTuple._2.isOperational()) {
+            AppConfig.debugLogger(
+                    getClass(),
+                    () -> String.format("Resolved model [%s] is not operational, avoiding its usage", modelName));
+            throw new DotAIModelNotOperationalException(String.format("Model [%s] is not operational", modelName));
+        }
+
+        final long sleep = lastRestCall.computeIfAbsent(aiModel, m -> 0L)
+                + aiModel.minIntervalBetweenCalls()
+                - System.currentTimeMillis();
+        if (sleep > 0) {
+            Logger.info(
+                    this,
+                    "Rate limit:"
+                            + aiModel.getApiPerMinute()
+                            + "/minute, or 1 every "
+                            + aiModel.minIntervalBetweenCalls()
+                            + "ms. Sleeping:"
+                            + sleep);
+            Try.run(() -> Thread.sleep(sleep));
```

Comment:
why do you need this sleep?

Reply:
No idea. It was already there since OpenAIRequest class.